src/object/C4Shape.cpp

```diff
@@ -25,180 +25,229 @@
+			new_wdt = fixtoi(rot_matrix[0] * Wdt + rot_matrix[1] * Hgt);
+			new_hgt = fixtoi(rot_matrix[1] * Wdt + rot_matrix[0] * Hgt);
+			x = new_x;
+			y = new_y - fixtoi(rot_matrix[1] * Wdt);
 		}
 		else
 		{
-			nwdt = fixtoi(mtx[0] * Wdt - mtx[1] * Hgt);
-			nhgt = fixtoi(- mtx[1] * Wdt + mtx[0] * Hgt);
-			x = nvtx + fixtoi(mtx[1] * Hgt);
-			y = nvty;
+			new_wdt = fixtoi(rot_matrix[0] * Wdt - rot_matrix[1] * Hgt);
+			new_hgt = fixtoi(- rot_matrix[1] * Wdt + rot_matrix[0] * Hgt);
+			x = new_x + fixtoi(rot_matrix[1] * Hgt);
+			y = new_y;
 		}
 	}
 	else
 	{
-		if (mtx[1] > 0)
+		if (rot_matrix[1] > 0)
 		{
-			nwdt = fixtoi(- mtx[0] * Wdt + mtx[1] * Hgt);
-			nhgt = fixtoi(mtx[1] * Wdt - mtx[0] * Hgt);
-			x = nvtx + fixtoi(mtx[0] * Wdt);
-			y = nvty - nhgt;
+			new_wdt = fixtoi(- rot_matrix[0] * Wdt + rot_matrix[1] * Hgt);
+			new_hgt = fixtoi(rot_matrix[1] * Wdt - rot_matrix[0] * Hgt);
+			x = new_x + fixtoi(rot_matrix[0] * Wdt);
+			y = new_y - new_hgt;
 		}
 		else
 		{
-			nwdt = fixtoi(- mtx[0] * Wdt - mtx[1] * Hgt);
-			nhgt = fixtoi(- mtx[1] * Wdt - mtx[0] * Hgt);
-			x = nvtx - nwdt;
-			y = nvty + fixtoi(mtx[0] * Hgt);
+			new_wdt = fixtoi(- rot_matrix[0] * Wdt - rot_matrix[1] * Hgt);
+			new_hgt = fixtoi(- rot_matrix[1] * Wdt - rot_matrix[0] * Hgt);
+			x = new_x - new_wdt;
+			y = new_y + fixtoi(rot_matrix[0] * Hgt);
 		}
 	}
-	Wdt = nwdt;
-	Hgt = nhgt;
+	Wdt = new_wdt;
+	Hgt = new_hgt;
 	if (Config.General.DebugRec)
 	{
-		rc.x=x; rc.y=y; rc.wdt=Wdt; rc.hgt=Hgt;
-		for (i=0; i<4; ++i)
-			{ rc.VtxX[i]=VtxX[i]; rc.VtxY[i]=VtxY[i]; }
+		C4RCRotVtx rc;
+		rc.x = x;
+		rc.y = y;
+		rc.wdt = Wdt;
+		rc.hgt = Hgt;
+		for (int32_t i = 0; i < 4; ++i)
+		{
+			rc.VtxX[i] = VtxX[i];
+			rc.VtxY[i] = VtxY[i];
+		}
 		AddDbgRec(RCT_RotVtx2, &rc, sizeof(rc));
 	}
 }
 
-void C4Shape::Stretch(int32_t iCon, bool bUpdateVertices)
+
+void C4Shape::Stretch(int32_t iCon, bool update_vertices)
 {
-	int32_t cnt;
-	x=x*iCon/FullCon;
-	y=y*iCon/FullCon;
-	Wdt=Wdt*iCon/FullCon;
-	Hgt=Hgt*iCon/FullCon;
-	if (bUpdateVertices)
-		for (cnt=0; cnt<VtxNum; cnt++)
+	x = ScaledByCon(x, iCon);
+	y = ScaledByCon(y, iCon);
+	Wdt = ScaledByCon(Wdt, iCon);
+	Hgt = ScaledByCon(Hgt, iCon);
+	if (update_vertices)
+	{
+		for (int32_t i = 0; i < VtxNum; i++)
 		{
-			VtxX[cnt]=VtxX[cnt]*iCon/FullCon;
-			VtxY[cnt]=VtxY[cnt]*iCon/FullCon;
+			VtxX[i] = ScaledByCon(VtxX[i], iCon);
+			VtxY[i] = ScaledByCon(VtxY[i], iCon);
 		}
+	}
 }
 
-void C4Shape::Jolt(int32_t iCon, bool bUpdateVertices)
+
+void C4Shape::Jolt(int32_t iCon, bool update_vertices)
 {
-	int32_t cnt;
-	y=y*iCon/FullCon;
-	Hgt=Hgt*iCon/FullCon;
-	if (bUpdateVertices)
-		for (cnt=0; cnt<VtxNum; cnt++)
-			VtxY[cnt]=VtxY[cnt]*iCon/FullCon;
+	y = ScaledByCon(y, iCon);
+	Hgt = ScaledByCon(Hgt, iCon);
+	if (update_vertices)
+	{
+		for (int32_t i = 0; i < VtxNum; i++)
+		{
+			VtxY[i] = ScaledByCon(VtxY[i], iCon);
+		}
+	}
 }
 
+
+inline bool C4Shape::ScaledByCon(int32_t value, int32_t con)
```

Comment:
This function doesn't require any members of C4Shape and can be made static.

Reply:
Actually, since it's not used outside of C4Shape.cpp at all, you can even make it a function that's local to C4Shape.cpp.

Reply:
How do I do that? Static is obvious, but I always get a compiler warning that the function is not used then (I am not C++ native ;))

Reply:
Remove it from the header and replace the declaration with
```cpp
static int32_t ScaledByCon(int32_t value, int32_t con)
```

Reply:
Yes, wow, I do not even know how I could miss that. Although, I wanted to make it inline, but the compiler cannot resolve the static inline function - OK, that was resolved by placing the function before the first use (had nothing to do with inline, just with the way C++ works)

Reply:
Also, I noticed that the collection zone (and likely the entrace, etc.) do not scale with Con. For structures that makes sense, but I think that for objects with StretchGrowth these should change with Con, too. Might be worth an issue in our bug tracker. Apart from that, the changes are done.